static/js/containers/DashboardPage_test.js

```diff
@@ -619,4 +621,40 @@ describe('DashboardPage', () => {
       });
     });
   });
+
+  describe("checkout", () => {
+    let dashboardResponse;
+    beforeEach(() => {
+      // Limit the dashboard response to 1 program
+      dashboardResponse = {"programs": [R.clone(DASHBOARD_RESPONSE.programs[0])]};
+    });
+
+    it('redirects to edX when the checkout API tells us to', () => {
+      let promise = Promise.resolve(EDX_CHECKOUT_RESPONSE);
+      let checkoutStub = helper.sandbox.stub(storeActions, 'checkout').returns(() => promise);
+      let course = makeCourse();
+      // Set all course runs to unpaid
+      course.runs = R.chain(R.set(R.lensProp('has_paid'), false), course.runs);
```

Comment:
`course` can be modified directly so you shouldn't need the lens or setter methods here, and it looks like you only need to modify one run in the course. Can you do something like: `course.runs[0] = { ...course.runs[0], status: STATUS_CAN_UPGRADE, course_end_date: moment().add(-1, 'months'), ` and so on